code/ndarray.c

```diff
@@ -163,8 +163,10 @@ void ndarray_fill_array_iterable(mp_float_t *array, mp_obj_t iterable) {
 
 #if ULAB_HAS_FUNCTION_ITERATOR
 size_t *ndarray_new_coords(uint8_t ndim) {
-    size_t *coords = m_new(size_t, ndim);
+    size_t *coords = m_new0(size_t, ndim);
+    #if !MICROPY_GC_CONSERVATIVE_CLEAR
     memset(coords, 0, ndim*sizeof(size_t));
+    #endif
```

Comment:
You don't need to use both `m_new9` and `MICROPY_GC_CONSERVATIVE_CLEAR`.  You can select one of these options:
```c
size_t *coords = m_new0(size_t, ndim);
```
or:
```c
size_t *coords = m_new(size_t, ndim);
#if !MICROPY_GC_CONSERVATIVE_CLEAR
memset(coords, 0, ndim*sizeof(size_t));
#endif
```

Have a look at the source code of `m_malloc0` to understand better.

The first option (just call `m_new0`) is much simpler and I suggest to do that.  The API of `m_new0` is unlikely to change.  But the macro `MICROPY_GC_CONSERVATIVE_CLEAR` may one day be removed.

Reply:
I have obviously tanked it completely. Thanks for pointing this out!